xskillscore/tests/test_contingency.py

```diff
@@ -122,3 +122,142 @@ def test_dichotomous_scores(dichotomous_Contingency, method, expected):
     """
     xs_score = getattr(dichotomous_Contingency, method)().item()
     npt.assert_almost_equal(xs_score, expected)
+
+
+@pytest.fixture
+def symmetric_edges():
+    """Category bin edges between 0 and 1."""
+    return np.linspace(-2, 2, 11)
+
+
+np.random.seed(42)
+
+
+@pytest.fixture
+def forecast_1d_long():
+    """Forecasts normally distributed around 0."""
+    s = 100
+    return xr.DataArray(np.random.normal(size=(s)), coords=[("time", np.arange(s))])
+
+
+@pytest.fixture
+def observation_1d_long():
+    """Observations normally distributed around 0."""
+    s = 100
+    return xr.DataArray(np.random.normal(size=(s)), coords=[("time", np.arange(s))])
+
+
+@pytest.mark.parametrize("drop_intermediate", [True, False])
+@pytest.mark.parametrize("chunk", [True, False])
+@pytest.mark.parametrize("input", ["Dataset", "DataArray"])
+@pytest.mark.parametrize(
+    "return_results", ["all_as_tuple", "area", "all_as_metric_dim"]
+)
+def test_roc_returns(
+    forecast_1d_long,
+    observation_1d_long,
+    symmetric_edges,
+    return_results,
+    input,
+    chunk,
+    drop_intermediate,
+):
+    """testing keywords and inputs"""
+    if "Dataset" in input:
+        name = "var"
+        forecast_1d_long = forecast_1d_long.to_dataset(name=name)
+        observation_1d_long = observation_1d_long.to_dataset(name=name)
+    if chunk:
+        forecast_1d_long = forecast_1d_long.chunk()
+        observation_1d_long = observation_1d_long.chunk()
+
+    roc(
+        forecast_1d_long,
+        observation_1d_long,
+        symmetric_edges,
+        dim="time",
+        drop_intermediate=drop_intermediate,
+        return_results=return_results,
+    )
+
+
+def test_roc_random_forecast(
+    forecast_1d_long, observation_1d_long, symmetric_edges, return_results="area"
+):
+    """Test that ROC around 0.5 for random forecast."""
+    area = roc(
+        forecast_1d_long,
+        observation_1d_long,
+        symmetric_edges,
+        dim="time",
+        return_results=return_results,
```

Comment:
If it's not parameterized could move the input to the function param `return_results=area`